sdk/nodejs/tests/runtime/langhost/cases/064.async_components/index.js

```diff
@@ -0,0 +1,38 @@
+// Test the ability to invoke provider functions via RPC.
+
+let assert = require("assert");
+let pulumi = require("../../../../../");
+
+class CustResource extends pulumi.CustomResource {
+    constructor(name, opts) {
+        super("test:index:CustResource", name, {}, opts)
+    }
+}
+
+class CompResource extends pulumi.ComponentResource {
+    constructor(name, opts) {
+        super("test:index:CompResource", name, {}, opts)
+        const data = this.getData();
+        this.a = pulumi.output(data.then(d => d.a));
+        this.b = pulumi.output(data.then(d => d.b));
+    }
+
+    /** @override */
+    async initialize() {
+        new CustResource("a", { parent: this });
+        new CustResource("b", { parent: this });
+        return { a: 1, b: 2 }
+    }
+}
+
+const comp = new CompResource("comp", {});
+comp.a.apply(v => {
```

Comment:
Note: once we've merged the changes to lift dependencies for outputs, we should update this test to check these somehow (e.g. by passing the results of these applies to another resource).

Reply:
Will do.